core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -282,6 +316,114 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with B
     assert(!failedTaskSet)
   }
 
+  test("scheduled tasks obey task and stage blacklists") {
```

Comment:
Is it possible to split the performance tests (that certain successful blacklist calls cause others not to be called) into a separate test from the correctness tests? This test is giant and as a result I'm worried about the complexity of debugging future failures.